Tests/NFUnitTest1/TestUuidUtilities.cs

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+using nanoFramework.TestFramework;
+using nanoFramework.Device.Bluetooth;
+
+namespace NFUnitTest1
+{
+  /// <summary>
+  /// Tests the different conversion and type detection functions for Bluetooth uuids
+  /// </summary>
+  [TestClass]
+  public class TestUuidUtilities
+  {
+    [TestMethod]
+    public void BaseUuidIsBluetoothSigUuid()
+    {
+      var baseUuid = new Guid("00000000-0000-1000-8000-00805f9b34fb");
+
+      Assert.IsTrue(Utilities.IsBluetoothSigUUID(baseUuid));
+
+      // the base uid marks the start of the pre-allocated range of 16bit or 32bit uuid values
+      // the 16bit or 32bit value is 0 so we give it the Uuid16 type
+      Assert.IsTrue(Utilities.TypeOfUuid(baseUuid) == Utilities.UuidType.Uuid16, "Expecting a 16bit uuid");
+    }
+
+    [TestMethod]
+    public void NonBaseUuidIs128Uuid()
+    {
+      var uuid = new Guid("00000000-0000-1000-8000-00805f9b34fc"); // last digit differs from base uuid
+
+      Assert.IsFalse(Utilities.IsBluetoothSigUUID(uuid));
+
+      // as this uid is not based on the base uid, the type is a random Uuid128
+      Assert.IsTrue(Utilities.TypeOfUuid(uuid) == Utilities.UuidType.Uuid128, "Expecting a 128bit uuid");
+    }
+
+    [TestMethod]
+    public void Test16BitUuids()
+    {
+      ushort value16 = 4660; // 0x1234
+
+      var serviceUid = Utilities.CreateUuidFromShortCode(value16);
+
+      var bytes = serviceUid.ToByteArray();
+      Assert.AreEqual((byte)52, bytes[0]); // 0x34
+      Assert.AreEqual((byte)18, bytes[1]); // 0x12
+      Assert.AreEqual((byte)0, bytes[2]);  // 0x00
+      Assert.AreEqual((byte)0, bytes[3]);  // 0x00
+
```

Comment:
It would look cleaner if you used hex values directly in code instead of in the comments
```
Assert.AreEqual((byte)0x34, bytes[0]);
```

Reply:
Agreed, Updated